lib/app-routes/common/auth/keys.js

```diff
@@ -40,6 +42,9 @@ exports.register = function(server, options, next) {
                     password: joi.binary().encoding('base64').required()
                 }
             },
+            response: {
+                schema: internals.credentialSchema
```

Comment:
Why is the credentials schema contained in the users controller?

I suggested using the `users` schema here as **part** of the credentials schema. Example:

```
schema: {
    username: joi.string().required,
    user: usersController.schema,
    id: joi.string().required,
    ...
}
```

Sorry if that wasn't clear.

Note: this will require that you move the validation schema currently in _lib/app-routes/common/users.js_ to the controller that you made in _lib/controllers/users.js_